internal/proxy/resource_ecs_proxy.go

```diff
@@ -0,0 +1,313 @@
+package proxy
+
+import (
+	"context"
+	"fmt"
+
+	integrationv1alpha1 "github.com/common-fate/sdk/gen/commonfate/control/integration/v1alpha1"
+
+	config_client "github.com/common-fate/sdk/config"
+	"github.com/common-fate/sdk/gen/commonfate/control/integration/v1alpha1/integrationv1alpha1connect"
+	"github.com/common-fate/sdk/service/control/integration"
+
+	"connectrpc.com/connect"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type ECSProxyModel struct {
+	ID           types.String `tfsdk:"id"`
+	AwsRegion    types.String `tfsdk:"aws_region"`
+	AwsAccountId types.String `tfsdk:"aws_account_id"`
+
+	ECSClusterName            types.String `tfsdk:"ecs_cluster_name"`
+	ECSTaskDefinitionFamily   types.String `tfsdk:"ecs_task_definition_family"`
+	ECSClusterReaderRoleARN   types.String `tfsdk:"ecs_cluster_reader_role_arn"`
+	ECSClusterSecurityGroupId types.String `tfsdk:"ecs_cluster_security_group_id"`
```

Comment:
Nit: casing Go struct fields should be PascalCase
```suggestion
	ECSClusterSecurityGroupID types.String `tfsdk:"ecs_cluster_security_group_id"`
```